src/opentime/timeRange.h

```diff
@@ -5,99 +5,246 @@
+                double selfStart = _start_time.value() / _start_time.rate();
+                double otherStart = other.value() / other.rate();
+                return fabs(otherStart - selfStart) <= epsilon;
+            }
+
+            /**finishes(other, epsilon)
+             * params: TimeRange other
+             * The start of self strictly antecedes the start of other by a value >= epsilon.
+             * The end of self strictly equals the end of other.
+             * The converse would be other.finishes(self)
+             */
+            bool finishes(TimeRange other, double epsilon = defaultEpsilon) const {
+                double selfStart = _start_time.value() / _start_time.rate();
+                double selfEnd = end_time_exclusive().value() / end_time_exclusive().rate();
+                double otherStart = other._start_time.value() / other._start_time.rate();
+                double otherEnd = other.end_time_exclusive().value() / other.end_time_exclusive().rate();
+                return fabs(selfEnd - otherEnd) <= epsilon && selfStart - otherStart >= epsilon;
+            }
+
+            /**finishes(other)
+             * params: RationalTime other
+             * The end of self strictly equals other.
+             */
+            bool finishes(RationalTime other, double epsilon = defaultEpsilon) const {
+                double selfEnd = end_time_exclusive().value() / end_time_exclusive().rate();
+                double otherEnd = other.value() / other.rate();
+                return fabs(selfEnd - otherEnd) <= epsilon;
+            }
+
+
+            /**operator equals()
+             * params: TimeRange lhs
+             *         TimeRange rhs
+             * The start of lhs strictly equals the start of rhs.
+             * The end of lhs strictly equals the end of rhs.
+             */
+            friend bool operator==(TimeRange lhs, TimeRange rhs) {
+                RationalTime start = RationalTime::absTime(lhs._start_time - rhs._start_time);
+                RationalTime duration = RationalTime::absTime(lhs._duration - rhs._duration);
+                return start.value() / start.rate() < defaultEpsilon &&
+                       duration.value() / duration.rate() < defaultEpsilon;
+            }
+
+            /**operator notequals()
+             * params: TimeRange lhs
+             *         TimeRange rhs
+             * Converse of equals() operator
+             */
+            friend bool operator!=(TimeRange lhs, TimeRange rhs) {
+                return !(lhs == rhs);
+            }
+
+            static TimeRange range_from_start_end_time(RationalTime start_time, RationalTime end_time_exclusive) {
+                return TimeRange{start_time,
+                                 RationalTime::duration_from_start_end_time(start_time, end_time_exclusive)};
+            }
+
+        private:
+            RationalTime _start_time, _duration;
+            static constexpr double defaultEpsilon = 1.0 / 1001.0;
```

Comment:
Couple notes:

1. I think if we're going to add a constant like this, we should specify the units, both by a comment and probably in the name of the variable itself, so lets use `defaultEpsilon_s`, to indicate that the epsilon is in unit seconds.
2. 1001.0 is probably not small enough to be useful for audio frame rates.  In order to make it useful in those contexts, where frame rates go considerably higher, after some discussion, @meshula and I concluded the value `1/(2.0*192000)` would be better, in order to cover audio rates up to 192khz.